### DISCUSSION TITLE
Cannot start quick installer (GitHub blocked in China)

### DISCUSSION THREAD

**Discussion by w45354** [RaspAP/raspap-webgui#1506] [General]
I follow the guide step by step, but when I type curl -sL https://install.raspap.com | bash, the quick installer cannot be launched

![d359266519f8b2de093fbdcd9c33461](https://github.com/RaspAP/raspap-webgui/assets/157502002/2e1fb92d-e1df-4d87-9275-578f146a3f3a)

I am in China, so I set the WLAN location to CN. And no VPN is set up in my RPI 4B. Normally, accessing foreign webs like GitHub is prohibited in China, I am wondering if this causes me to get no access to raspap's server, leading to a failure to launch a quick installer.

![d359266519f8b2de093fbdcd9c33461](https://github.com/RaspAP/raspap-webgui/assets/157502002/72f80ee3-3352-4c51-b630-3cca8d6493cd)

**Comment by billz** [comment]
See https://github.com/RaspAP/raspap-webgui/discussions/1492#discussioncomment-8021446

**Comment by w45354** [comment]
I cannot get access to Raspap's server because I am in China, and I am trying to install it manually I also cannot connect to the service as github is blocked in China. do you have any solutions if I would like to install RaspAP in china?
![533719ff1e3cfc009865cd09069dfd5](https://github.com/RaspAP/raspap-webgui/assets/157502002/b69a9570-df53-407b-9fb1-6a47127f45f4)

**Comment by billz** [comment]
You could run the installer locally, for example `./installers/raspbian.sh` but this first requires that you have the project from GitHub. If you are unable to access the installer due to it being blocked, you won't be able to clone files from GitHub. 

According to our stats, China is the third most popular country for RaspAP installs. So users must be able to obtain it some other way. Have you tried a GitHub mirror?

**Comment by w45354** [comment]
I can download files from GitHub from my PC as a VPN client is installed. Now I already have the project files in my RPI, but I don't know how to set it up by the command codings, do you have any guidelines?

**Reply by billz** [reply]
OK, I understand now. The installer needs to download the project from GitHub. If it's unable to, it will stop and report an error. Since you already have the files, your could follow the [manual installation](https://docs.raspap.com/manual/) steps. 

Another solution is to deploy RaspAP in a [Docker container](https://docs.raspap.com/docker/).

**Comment by w45354** [comment]
couldn't find /config/defaut_hostapd.conf
what is the problem with this hint

**Reply by billz** [reply]
> Optionally, backup your existing hostapd.conf:

This is optional, only if you have an existing hostapd.conf. If not, it's safe to skip this.

**Comment by hxzhao527** [comment]
![image](https://github.com/RaspAP/raspap-webgui/assets/8402502/566a125a-f640-49e9-9a76-7f2447228b9e)

no any output after execute the installer.sh

**Reply by hxzhao527** [reply]
I got the reason. https://github.com/RaspAP/raspap-webgui/blob/6541e9b2e1ee9f064b82e198be698f32b9848cb0/installers/raspbian.sh#L306 Because `wget` with `-q`.

![image](https://github.com/RaspAP/raspap-webgui/assets/8402502/9f0d9e44-2e42-447f-b12e-3560fdb688c1)
Still because of network problem.

**Comment by billz** [comment]
For others who are unable to access GitHub due to China's Great Firewall (GFW; 防火长城) @hxzhao527 created [this fork](https://github.com/hxzhao527/raspap-webgui/commit/c58b0cac86e4b6d96c13acdc5765d24dc22a3a3e) which uses `https://mirror.ghproxy.com/`.

**Comment by billz** [comment]
Update: according to GreatFire ghproxy.com is 100% blocked in China now. This is no longer a viable option.
https://en.greatfire.org/https/ghproxy.com

**Comment by frankozland** [comment]
proton offers free vpn for a single user.... have you tried that?
https://protonvpn.com/free-vpn

or tor? https://medium.com/@brotheralameen/how-to-use-tor-without-a-tor-browser-6d9034dfcc58
https://www.torproject.org/download/tor/

https://support.torproject.org/censorship/connecting-from-china/
https://www.comparitech.com/blog/vpn-privacy/whats-the-best-vpn-for-china-5-that-still-work-in-2016/

**Reply by brotheralameen1** [reply]
Interesting to see my article getting promoted. Thanks.

**Comment by frankozland** [comment]
some other ideas from reddit
https://www.reddit.com/r/dumbclub/comments/1d64bh4/howdy/

Try freebrowser.org

 https://psiphon.ca/ (created by CitizenLab which is affiliated with the University of Toronto, but now seems to be its own company)

https://www.nthlink.com/ (Funded by Google Jigsaw and the Open Technology Fund).

if willing to pay can try setting up  own Outline node https://getoutline.org/en-GB/, the setup is very user friendly.
